src/date/expand-pattern.js

```diff
@@ -33,10 +55,30 @@ return function( pattern, cldr ) {
 
 	switch ( true ) {
 		case "skeleton" in pattern:
+			skeleton = pattern.skeleton;
 			result = cldr.main([
 				"dates/calendars/gregorian/dateTimeFormats/availableFormats",
-				pattern.skeleton
+				skeleton
 			]);
+			if ( result === undefined ) {
+				for ( i = 0 ; i < skeleton.length ; i++ ) {
+					if ( /[hHms]/.test( skeleton[i] ) ) {
+						timeSkeleton += skeleton[i];
+					} else if ( /[GyQMEd]/.test( skeleton[i] ) ) {
+						dateSkeleton += skeleton[i];
+					}
+				}
```

Comment:
Instead of the whole for loop, you could:

```js
timeSkeleton = skeleton.replace( /[^hHms]/g, "" );
```

Analogous thing for dateSkeleton.